### PR TITLE
[UserNotifications] Updates for Xcode13 beta 3

### PR DESCRIPTION
**Pull request by tj-devel709**

### CODE REVIEW

File: src/usernotifications.cs
Comment (rolfbjarne) [inline src/usernotifications.cs:458]: Methods should be named with a verb, so something like this:

```suggestion
		UNNotificationContent Update (IUNNotificationContentProviding fromProvider, [NullAllowed] out NSError outError);
```

This is very different, but it's how it's named in Swift (except that it's called `updating` in Swift, which isn't the verb tense we use)

File: src/usernotifications.cs
Comment (rolfbjarne) [inline src/usernotifications.cs:180]: Really minor, but you have two newlines here:

```suggestion

```

**Review by rolfbjarne — APPROVED** [review]

File: src/usernotifications.cs
Comment (mandel-macaque) [inline src/usernotifications.cs:204]: Apple, the gift that keeps on giving, only Catalyst??

**Review by mandel-macaque — APPROVED** [review]
Approving, please address the two comments from rolf AND update the WIKI with the PR!

**Comment by mandel-macaque** [comment]
DO NOT MERGE, changes are coming with xcode13 beta 3 :/

File: src/usernotifications.cs
Comment (tj-devel709) [reply src/usernotifications.cs:204]: Yup lol it seems so

File: src/usernotifications.cs
Comment (spouliot) [inline src/usernotifications.cs:440]: Wrong casing: our binding is `SummaryArgument`

File: src/usernotifications.cs
Comment (spouliot) [inline src/usernotifications.cs:445]: same (wrong casing)

File: src/usernotifications.cs
Comment (spouliot) [inline src/usernotifications.cs:466]: that should be mapped to the catalyst-only enum you bound earlier

File: src/usernotifications.cs
Comment (spouliot) [inline src/usernotifications.cs:519]: wrong casing

File: src/usernotifications.cs
Comment (spouliot) [inline src/usernotifications.cs:524]: wrong casing

File: src/usernotifications.cs
Comment (spouliot) [inline src/usernotifications.cs:545]: 👍

File: src/usernotifications.cs
Comment (spouliot) [reply src/usernotifications.cs:440]: There's not much value in naming the property... and it makes the assembly larger
Simply use `"This property is ignored."`

File: src/usernotifications.cs
Comment (spouliot) [inline src/usernotifications.cs:919]: camel casing for parameters -> `imageName`

File: src/usernotifications.cs
Comment (spouliot) [inline src/usernotifications.cs:923]: camel casing for parameters -> `imageName`

File: tests/xtro-sharpie/tvOS-UserNotifications.ignore
Comment (spouliot) [inline tests/xtro-sharpie/tvOS-UserNotifications.ignore:7]: why were the deprecation attributes were only added for iOS ?

File: tests/xtro-sharpie/watchOS-UserNotifications.ignore
Comment (spouliot) [inline tests/xtro-sharpie/watchOS-UserNotifications.ignore:7]: why were the deprecation attributes were only added for iOS ?

**Review by spouliot — CHANGES_REQUESTED** [review]

File: tests/xtro-sharpie/tvOS-UserNotifications.ignore
Comment (tj-devel709) [reply tests/xtro-sharpie/tvOS-UserNotifications.ignore:7]: I only added the deprecation attribute to iOS because TV and Watch was not supported in these selectors. I thought this would be better than introducing them and deprecating them in this update. What do you think?

File: tests/xtro-sharpie/watchOS-UserNotifications.ignore
Comment (tj-devel709) [reply tests/xtro-sharpie/watchOS-UserNotifications.ignore:7]: Copying and pasting this here:
I only added the deprecation attribute to iOS because TV and Watch was not supported in these selectors. I thought this would be better than introducing them and deprecating them in this update. What do you think?

File: tests/xtro-sharpie/tvOS-UserNotifications.ignore
Comment (spouliot) [reply tests/xtro-sharpie/tvOS-UserNotifications.ignore:7]: make sense :)
In such cases you can add a comment like `# both added and deprecated in Xcode 13 beta 1`
It helps reviews and can be useful in the future (where the wiki data might not be available anymore)

Comment (tj-devel709) [reply tests/xtro-sharpie/tvOS-UserNotifications.ignore:7]: Makes sense! I'll update it :)

File: src/usernotifications.cs
Comment (tj-devel709) [reply src/usernotifications.cs]: Updated it! Then I saw B3 removed the enum type and these properties :(

File: src/usernotifications.cs
Comment (tj-devel709) [reply src/usernotifications.cs]: Lol B3 removed this!

**Review by spouliot — APPROVED** [review]
don't forget to update the PR title to beta 3 before merging

File: src/usernotifications.cs
Comment (mandel-macaque) [reply src/usernotifications.cs]: I suspected something of the kind ;)

**Review by mandel-macaque — APPROVED** [review]

File: tests/xtro-sharpie/watchOS-UserNotifications.ignore
Comment (mandel-macaque) [inline tests/xtro-sharpie/watchOS-UserNotifications.ignore:1]: @tj-devel709  do we get these errors when using beta 3? Have you tried removing it?

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

File: tests/xtro-sharpie/watchOS-UserNotifications.ignore
Comment (tj-devel709) [reply tests/xtro-sharpie/watchOS-UserNotifications.ignore:1]: When I remove this, I get errors still!

**Review by mandel-macaque — APPROVED** [review]

**Comment by tj-devel709** [comment]
Updating to fix this unrelated issue: https://github.com/xamarin/maccore/issues/2472

**Comment by tj-devel709** [comment]
Failing unrelated tests: https://github.com/xamarin/maccore/issues/2443 and 
https://github.com/xamarin/maccore/issues/2471

**Pull request merged**
